constraints/cpu.md

```diff
@@ -0,0 +1,35 @@
+In general, we recommend that you do not edit this variable, unless you know what are you doing. If your container needs more than one vCPU, use [docker `--cpus`][1], [docker update][4] or [kubernetes `cpu` limits][2] to assign more resources.
+
+## Virtual CPUs and Kubernetes PODs
+
+A kubernetes POD is a group of one or more containers, with shared storage and network, and a specification for how to run the containers [[1][3]]. In Kata Containers this group of containers runs inside the same virtual machine. If you do not specify a CPU constraint, the runtime does not hot add more vCPUs and the container is not placed inside a CPU cgroup.  Instead, the container uses the number of vCPUs specified by `default_vcpus` and shares these resources with other containers in the same situation (without a CPU constraint).
+
+## Containers life cycle
+
+When you create a container with a CPU constraint, the runtime hot adds the number of vCPUs required by the container. Similarly, when the container stops, the runtime hot removes these resources.
+
+## Containers without CPU constraint
+
+A container without a CPU constraint uses the default number of vCPUs specified in the configuration file. In the case of Kubernetes PODs, containers without a CPU constraint use and share between them the default number vCPUs. For example, if default_vcpus is equal to 1 and you have 2 containers without CPU constraints with each container trying to consume 100% of vCPU, the resources divide in two parts, 50% of vCPU for each container because your virtual machine does not have enough resources to satisfy containers needs. If you wish to give access to a greater or lesser portion of vCPUs to a specific container, you can use [docker --cpu-shares][1] or [kubernetes `cpu` requests][2].
+Before running containers without CPU constraint, consider that your containers are not running alone. Since your containers run inside a virtual machine other processes try to use the vCPUs as well (e.g. `systemd` and the Kata Container [agent][5]). In general, we recommend setting `default_vcpus` equal to 1 to allow non-container processes to run on this vCPU and to specify a CPU constraint for each container. If your container is already running and needs more vCPUs, you can add more using [docker update][4].
+
+## Containers with CPU constraint
+
+The runtime calculates the number of vCPUs required by containers with CPU constraint using the following formula: `(quota + (period -1)) / period`. The result determines the number of vCPU to hot plug into the virtual machine. Once the vCPUs have been hot added, the [agent][5] places the container inside a CPU cgroup. This placement allows the container to use only its assigned resources.
```

Comment:
Can you provide explanation of quota and period here, what they mean and where they are assigned for docker and k8s.